lib/commands/os/configure.ts

```diff
@@ -225,6 +228,7 @@ export default class OsConfigureCmd extends Command {
 			(await getOsVersionFromImage(params.image, deviceTypeManifest, devInit));
 
 		answers.provisioningKeyName = options['provisioning-key-name'];
+		answers.developmentMode = options.dev;
```

Comment:
Setting `answers.developmentMode` here causes this option to be eventually passed to:
`sdk.models.os.getConfig(fleetSlug, { developmentMode: true/false, ... })`
(in `lib/utils/config.ts` that did not need to be modified by this PR).

I have tested it, it seems to work OK.

Reply:
This should probably also fail if the version provided/resolved ends up being a prod one.

Reply:
Similarly w/ `config generate`, this is also enough b/c it ends up reaching `/download-config`.